ksqldb-engine/src/main/java/io/confluent/ksql/planner/LogicalPlanner.java

```diff
@@ -755,25 +757,34 @@ private static void verifyStreamTableJoin(final JoinInfo joinInfo, final PlanNod
       ));
     }
 
-    if (!(leftExpression instanceof ColumnReferenceExp)) {
-      throw new KsqlException(String.format(
-          "Invalid join condition:"
-              + " foreign-key table-table joins with expressions are not supported yet."
-              + " Got %s = %s.",
-          joinInfo.getFlippedLeftJoinExpression(),
-          joinInfo.getFlippedRightJoinExpression()
-      ));
-    }
-
-    // we need to extend this to support expressions later on
-    final ColumnReferenceExp fkColumnReference = (ColumnReferenceExp) leftExpression;
+    final CodeGenRunner codeGenRunner = new CodeGenRunner(
+        leftNode.getSchema(),
+        ksqlConfig,
+        metaStore
+    );
 
-    final SqlType fkColumnType =
-        leftNode.getSchema().findColumn(fkColumnReference.getColumnName()).get().type();
+    final VisitParentExpressionVisitor<Optional<Expression>, Context<Void>> unqualifiedRewritter =
+        new VisitParentExpressionVisitor<Optional<Expression>, Context<Void>>(Optional.empty()) {
+          @Override
+          public Optional<Expression> visitQualifiedColumnReference(
+              final QualifiedColumnReferenceExp node,
+              final Context<Void> ctx
+          ) {
+            return Optional.of(new UnqualifiedColumnReferenceExp(node.getColumnName()));
```

Comment:
The `ExpressionEvaluator` cannot handle fully qualified column references, thus we rewrite the given expression accordingly.

Reply:
I had a sync with @mjsax offline about this specifically, and our thoughts are that since the codeGenRunner is used for physical plan generation as well at which stage there's no qualified column refs since we would just prefix the names as `table/stream__column`, is that right @vcrfxia ??

If yes, just as a general note here I'm feeling that we should use a separate class than codeGenRunner for traversals in the logical/physical plan generation respectively, than reusing it, since for logical planning all we really need are return type / param types etc, while nevertheless today we always run the same full-fledged traversal multiple times at different stages. Is my understanding reasonable?